template.tpl

```diff
@@ -0,0 +1,2025 @@
+const JSON = require('JSON');
+const Math = require('Math');
+const sendHttpRequest = require('sendHttpRequest');
+
+// Helpers
+const cleanObject = (obj) => {
+  let target = {};
+
+  for (let prop in obj) {
+    if (obj.hasOwnProperty(prop) && obj[prop] != null) {
+      target[prop] = obj[prop];
+    }
+  }
+
+  return target;
+};
+
+const merge = (args) => {
+  let target = {};
+
+  const addToTarget = (obj) => {
+    for (let prop in obj) {
+      if (obj.hasOwnProperty(prop)) {
+        target[prop] = obj[prop];
+      }
+    }
+  };
+
+  for (let i = 0; i < args.length; i++) {
+    addToTarget(args[i]);
+  }
+
+  return target;
+};
+
+/*
+ * Replaces all occurences of a substring in a string.
+ *
+ * @param str {string} - the string to replace into
+ * @param substr {string} - the substring to replace
+ * @param newSubstr {string} - the replacement substring
+ * @returns - the string with all occurences of substr replaced with newSubstr
+ */
+const replaceAll = (str, substr, newSubstr) => {
+  let finished = false,
+    result = str;
+  while (!finished) {
+    const newStr = result.replace(substr, newSubstr);
+    if (result === newStr) {
+      finished = true;
+    }
+    result = newStr;
+  }
+  return result;
+};
+
+/*
+ * Returns whether a string is upper case.
+ *
+ * @param value {string} - the string to check
+ * @returns - boolean
+ */
+const isUpper = (value) => {
+  return value === value.toUpperCase() && value !== value.toLowerCase();
+};
+
+/*
+ * Converts a string to snake case.
+ *
+ * @param value {string} - the string to convert
+ * @returns - the converted string
+ */
+const toSnakeCase = (value) => {
+  let result = '';
+  let previousChar;
+  for (var i = 0; i < value.length; i++) {
+    let currentChar = value.charAt(i);
+    if (isUpper(currentChar) && i > 0 && previousChar !== '_') {
+      result = result + '_' + currentChar;
+    } else {
+      result = result + currentChar;
+    }
+    previousChar = currentChar;
+  }
+  return result;
+};
+
+/*
+ * Left-pads an integer with zeros. Assumes a non-negative integer.
+ *
+ * @param x {integer} - the integer to pad
+ * @param totalLen {integer} - the desired total length
+ * @returns - the resulting string after padding with zeros
+ */
+function pad(x, totalLen) {
+  const s = x.toString();
+  if (s.length < totalLen) {
+    return pad('0' + s, totalLen);
+  }
+  return s;
+}
+
+/*
+ * Given the time components, returns the ISO-8601 time code.
+ *
+ */
+function formatISO(yr, mon, day, hrs, mins, secs, msecs) {
+  const isoDate = [yr.toString(), pad(mon, 2), pad(day, 2)].join('-');
+  const isoTime = [pad(hrs, 2), pad(mins, 2), pad(secs, 2)].join(':');
+
+  return isoDate + 'T' + isoTime + '.' + pad(msecs, 3) + 'Z';
+}
+
+/*
+ * Transforms a unix timestamp in milliseconds to ISO-8601 time code.
+ * Ref: https://howardhinnant.github.io/date_algorithms.html#civil_from_days
+ */
+function unixMillisToISO(unixMillis) {
```

Comment:
This is fun 😬 

Looks good though!

Reply:
Indeed, it was standard-JavaScript fun! Unit tested it as much as i could and seems consistent so far..